### PR TITLE
[EBPF-492] Show line-by-line complexity information

### PR DESCRIPTION
**Pull request by gjulianm**
### What does this PR do?

This PR adds a new task `ebpf.annotate-complexity` that shows source code annotated with the complexity information extracted from the verifier.

The information is extracted using the same Go program already created to print verification stats. In order to support this feature and possible future ones, the following changes have been made:

* Summary file is written to JSON directly from the Go program
* The Go program now can write the verifier log to files, allowing easy debugging.
* Source code information is retrieved using `llvm-objdump` (Go's `dwarf` library does not support correctly the DWARF format for eBPF programs. 
* Complexity information is reconstructed from the verifier log and source code maps, and written in separate JSON files (one per program, as those JSON files can get very big).

In the Python tasks, the following changes have been made:
* Separate the previous `ebpf.print-verification-stats` into two tasks, one that collects the information from the Go program (`collect-verification-stats`) and another that prints them (`print-verification-stats`, as before). The latter will call the former if the JSON file with the data is not created.
* Create a new task `ebpf.annotate-complexity` that receives a binary name and program name, and prints annotated source code. Optionally it can show the assembly instructions related to each call, when using the `--show-assembly` flag.

Example result:

Screenshot 2024-04-19 at 16 22 21

A legend is printed automatically with the task, explaining the numbers and color codes.

Example with assembly: 

Screenshot 2024-04-19 at 17 07 48

### Motivation

The idea of this PR is to have a first proof of concept around using the verifier log to show complexity, in a way that's easy to consume for eBPF developers.

### Possible Drawbacks / Trade-offs

Right now, the output format is not especially useful to navigate the code and see the data in context. However, given the diversity of IDEs/editors used by eBPF developers, it seems a first approach using a terminal application is the best way to start.

### Describe how to test/QA your changes

Locally tested.

### CODE REVIEW

File: tasks/ebpf.py [hunk 1]
```diff
@@ -159,3 +207,99 @@ def print_verification_stats(
...
+    if colored is None:
+        raise Exit("termcolor is required to print colored output")
+
+    print("\n\n")
+    print("## Complexity legend ##")
+    print("For each source line the verifier processed, we output:")
+    print("  · A code [Ii|Pp|Tt] where:")
+    print("    - Ii is the number of eBPF instructions generated by this source line")
+    print("    - Pp is the number of passes the verifier made over assembly generated by this source line")
+    print("    - Tt is the total number of instructions processed by the verifier for this source line")
```
Comment (brycekahle) [inline tasks/ebpf.py:222]: Do we ever use the uppercase versions?

Comment (gjulianm) [reply tasks/ebpf.py:222]: The uppercase is meant to represent the number that gets placed but I see how that's confusing. Let me change it.

Comment (brycekahle) [reply tasks/ebpf.py]: It looks like you kept the uppercase version, but the lowercase one is used in the output. Make sure to change the line starting with `A code` too.

Comment (gjulianm) [reply tasks/ebpf.py]: Done, let me know if it looks better now.

**Review by chouetz — APPROVED** [review]
ok for agent-ci-experience files

**Review by brycekahle — APPROVED** [review]

**Pull request merged**